src/daemon/db_util.rs

```diff
@@ -121,16 +126,62 @@ pub async fn import_chain_as_forest_car(
                     .await?;
                 forest_car_db_temp_path.persist(&forest_car_db_path)?;
             }
+            anyhow::Ok(())
+        }
+    };
+
+    match import_mode {
+        ImportMode::Auto => {
+            if Url::parse(&from_path.display().to_string()).is_ok() {
+                // Fallback to move if from_path is url
+                move_or_copy(ImportMode::Move).await?;
+            } else if ForestCar::is_valid(&EitherMmapOrRandomAccessFile::open(from_path)?) {
+                tracing::info!(
+                    "Hardlinking {} to {}",
+                    from_path.display(),
+                    forest_car_db_path.display()
+                );
+                if std::fs::hard_link(from_path, &forest_car_db_path).is_err() {
+                    tracing::warn!("Error creating hardlink, fallback to copy");
+                    move_or_copy(ImportMode::Copy).await?;
+                }
+            } else {
+                tracing::warn!(
+                    "Snapshot file is not a valid forest.car.zst file, fallback to copy"
+                );
+                move_or_copy(ImportMode::Copy).await?;
+            }
+        }
+        ImportMode::Copy | ImportMode::Move => {
+            move_or_copy(import_mode).await?;
         }
         ImportMode::Symlink => {
             let from_path = std::path::absolute(from_path)?;
             if ForestCar::is_valid(&EitherMmapOrRandomAccessFile::open(&from_path)?) {
+                tracing::info!(
+                    "Symlinking {} to {}",
+                    from_path.display(),
+                    forest_car_db_path.display()
+                );
                 std::os::unix::fs::symlink(from_path, &forest_car_db_path)
                     .context("Error creating symlink")?;
             } else {
                 bail!("Snapshot file must be a valid forest.car.zst file");
             }
         }
+        ImportMode::Hardlink => {
```

Comment:
Can we add tests to different snapshot import modes in `src/test`?

Reply:
Unit tests updated to cover `auto` and `hardlink` cases

Reply:
Lovely! I knew I created them somewhere but couldn't find them. :rofl: